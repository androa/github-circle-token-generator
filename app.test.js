const request = require("supertest");
const fs = require("fs");
const { setGitHubTokenOnProject } = require("./set-github-token-on-project");

jest.mock("./set-github-token-on-project");

process.env.GITHUB_APP_ID = "123";
process.env.GITHUB_APP_KEY = fs.readFileSync("__mocks__/github-app-key.pem");

const app = require("./app");

test("token gets added to environment of project", done => {
  setGitHubTokenOnProject.mockResolvedValue({ name: "BAR", value: "xxxFOO" });

  request(app)
    .get("/token/inject?username=test&project=test")
    .expect(({ text }) => {
      console.log(text);
      expect(text).toMatchSnapshot();
    })
    .expect(200, done);
});

test("request fails if username is missing", done => {
  request(app)
    .get("/token/inject?project=test")
    .expect(({ body }) => {
      expect(body).toMatchSnapshot();
    })
    .expect(400, done);
});

test("request fails if project is missing", done => {
  request(app)
    .get("/token/inject?username=test")
    .expect(({ body }) => {
      expect(body).toMatchSnapshot();
    })
    .expect(400, done);
});

test("request fails if username and project is missing", done => {
  request(app)
    .get("/token/inject")
    .expect(({ body }) => {
      expect(body).toMatchSnapshot();
    })
    .expect(400, done);
});
