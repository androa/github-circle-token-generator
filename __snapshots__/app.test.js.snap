// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request fails if project is missing 1`] = `
Object {
  "errors": Array [
    Object {
      "location": "query",
      "msg": "Project identifier is required",
      "param": "project",
    },
  ],
}
`;

exports[`request fails if username and project is missing 1`] = `
Object {
  "errors": Array [
    Object {
      "location": "query",
      "msg": "Username is required",
      "param": "username",
    },
    Object {
      "location": "query",
      "msg": "Project identifier is required",
      "param": "project",
    },
  ],
}
`;

exports[`request fails if username is missing 1`] = `
Object {
  "errors": Array [
    Object {
      "location": "query",
      "msg": "Username is required",
      "param": "username",
    },
  ],
}
`;

exports[`token gets added to environment of project 1`] = `"Successfully created token (ending with xxxFOO) and injected it to CircleCI build (test/test)"`;
